pkg/multicast/multicast.go

```diff
@@ -691,24 +695,24 @@ func (c *Client) setupConnections() ([]net.IP, error) {
 		portIPsMap[port] = ips
 	}
 
-	var ipAddrs []net.IP
+	portNetIPsMap := make(map[int][]net.IP)
 	for port, ips := range portIPsMap {
 		newIPs, err := c.setupConnection(port, ips)
 		if err != nil {
 			c.log.Errorw("Fail to setup connection", "port", port, "ips", ips)
 			return nil, err
 		}
 
-		ipAddrs = append(ipAddrs, newIPs...)
+		portNetIPsMap[port] = append(portNetIPsMap[port], newIPs...)
```

Comment:
I think go doesn't allow for directly appending to an element of map

Reply:
> I think go doesn't allow for directly appending to an element of map

It works, I tested it. But the map should be 1 port - 1 connection. I will change it
<img width="456" alt="image" src="https://user-images.githubusercontent.com/34814121/205609317-031046ce-b311-441f-b175-b9933127bb3a.png">